### ISSUE TITLE
File write permissions are not respected

### ISSUE THREAD

**Issue by nyxkn** [musescore/MuseScore#20393]
### Issue type

Opening/saving issue

### Bug description

The following applies for Linux. Not tested on other platforms.

A Musescore file marked read-only will regain the write permission when saving it in Musescore, and will save successfully.

This is not the typical behaviour for applications.
Instead, I would expect Musescore to respect the missing write permission and not go through with the save. An alert about the file being read-only would probably be ideal, but even a silent failure would be more appropriate than overriding the permissions.

Related, saving also restores the owner and group to the current user's owner and group. This is also rather odd behaviour for an application.

tl;dr, saving overwrites all permission properties of a file. This is odd and undesirable behaviour.

### Steps to reproduce

Remove write permisson of a musescore file
Open file in musescore, edit, save
File is successfully saved and write permission restored (!)

### MuseScore Version

4.1.1

### Regression

I don't know

### Operating system

Archlinux

**Comment by oktophonie** [comment]
Duplicate of https://github.com/musescore/MuseScore/issues/18681
